core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java

```diff
@@ -59,16 +60,24 @@
 
 @ExtendWith(ParameterizedTestExtension.class)
 public class TestRemoveSnapshots extends TestBase {
+
   @Parameter(index = 1)
   private boolean incrementalCleanup;
 
-  @Parameters(name = "formatVersion = {0}, incrementalCleanup = {1}")
+  @Parameter(index = 2)
+  private boolean retainDataFile;
+
+  @Parameters(name = "formatVersion = {0}, incrementalCleanup = {1}, retainDataFile = {2}")
   protected static List<Object> parameters() {
     return Arrays.asList(
-        new Object[] {1, true},
-        new Object[] {2, true},
-        new Object[] {1, false},
-        new Object[] {2, false});
+        new Object[] {1, true, false},
```

Comment:
Independent of the approach we take, I'm not sure we really need to parameterize here in the tests. I look at it as just having separate tests, one per cleanup mode or one per if data files are retained or not. I don't think it's super valuable to test that for every possible combination of expire snapshots because we're really just testing one dimension of the API for this change independent of which snapshots were removed.